### ISSUE TITLE
Has trouble with binary values: `sha256: b'...'`

### ISSUE THREAD

**Issue by huyz** [mnrkbys/bgiparser#1]
I guess the `sha256` binary values need to be encoded.

```
Traceback (most recent call last):
  File "/Users/huyz/git/+dot/bgiparser/./bgiparser.py", line 240, in <module>
    main()
  File "/Users/huyz/git/+dot/bgiparser/./bgiparser.py", line 226, in main
    print(json.dumps(login_items, ensure_ascii=False, indent=4))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/huyz/.pyenv/versions/3.11.5/lib/python3.11/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/Users/huyz/.pyenv/versions/3.11.5/lib/python3.11/json/encoder.py", line 202, in encode
    chunks = list(chunks)
             ^^^^^^^^^^^^
  File "/Users/huyz/.pyenv/versions/3.11.5/lib/python3.11/json/encoder.py", line 432, in _iterencode
    yield from _iterencode_dict(o, _current_indent_level)
  File "/Users/huyz/.pyenv/versions/3.11.5/lib/python3.11/json/encoder.py", line 406, in _iterencode_dict
    yield from chunks
  File "/Users/huyz/.pyenv/versions/3.11.5/lib/python3.11/json/encoder.py", line 326, in _iterencode_list
    yield from chunks
  File "/Users/huyz/.pyenv/versions/3.11.5/lib/python3.11/json/encoder.py", line 406, in _iterencode_dict
    yield from chunks
  File "/Users/huyz/.pyenv/versions/3.11.5/lib/python3.11/json/encoder.py", line 439, in _iterencode
    o = _default(o)
        ^^^^^^^^^^^
  File "/Users/huyz/.pyenv/versions/3.11.5/lib/python3.11/json/encoder.py", line 180, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type bytes is not JSON serializable
```

**Comment by mnrkbys** [comment]
Thanks for sharing the problem.
Actually, I don't have samples that have sha256 with bytes type, so could you share your BackgroundItems-v*.btm?

**Comment by huyz** [comment]
I think this one of the plist's relevant snippets:

```xml
                <string>org.pqrs.service.agent.Karabiner-Menu</string>
                <dict>
                        <key>$class</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>100</integer>
                        </dict>
                        <key>NS.base</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3740</integer>
                        </dict>
                        <key>NS.relative</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3741</integer>
                        </dict>
                </dict>
                <dict>
                        <key>$class</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>100</integer>
                        </dict>
                        <key>NS.base</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>0</integer>
                        </dict>
                        <key>NS.relative</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>2115</integer>
                        </dict>
                </dict>
                <string>Contents/Library/LaunchAgents/org.pqrs.service.agent.Karabiner-Menu.plist</string>
                <data>
                qYt36M+G9186LAEGEV083X3O9cPCozB0gXCGZN5nqcU=
                </data>
                <string>/Library/Application Support/org.pqrs/Karabiner-Elements/Karabiner-Menu.app/Contents/MacOS/Karabiner-Menu</string>
                <dict>
                        <key>$class</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>18</integer>
                        </dict>
                        <key>NS.objects</key>
                        <array>
                                <dict>
                                        <key>CF$UID</key>
                                        <integer>3745</integer>
                                </dict>
                        </array>
                </dict>
                <string>/Library/Application Support/org.pqrs/Karabiner-Elements/Karabiner-Menu.app/Contents/MacOS/Karabiner-Menu</string>
                <string>Karabiner-Menu</string>
                <string>Fumihiko Takayama</string>
                <string>G43BCU2T37</string>
                <data>
                cIGZAgEBsIGTMAkMBGNjYXQCAQAwCQwEY29tcAIBATBwDARyZXFzsGgwLQwS
                c2lnbmluZy1pZGVudGlmaWVyDBdvcmcucHFycy5LYXJhYmluZXItTWVudTAd
                DA90ZWFtLWlkZW50aWZpZXIMCkc0M0JDVTJUMzcwGAwTdmFsaWRhdGlvbi1j
                YXRlZ29yeQIBBjAJDAR2ZXJzAgEB
                </data>
                <dict>
                        <key>$class</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>18</integer>
                        </dict>
                        <key>NS.objects</key>
                        <array/>
                </dict>
                <dict>
                        <key>$class</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>19</integer>
                        </dict>
                        <key>associatedBundleIdentifiers</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>0</integer>
                        </dict>
                        <key>bookmark</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>0</integer>
                        </dict>
                        <key>bundleIdentifier</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>0</integer>
                        </dict>
                        <key>container</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3681</integer>
                        </dict>
                        <key>developerName</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3762</integer>
                        </dict>
                        <key>disposition</key>
                        <integer>11</integer>
                        <key>executableModificationDate</key>
                        <real>744853749</real>
                        <key>executablePath</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3758</integer>
                        </dict>
                        <key>flags</key>
                        <integer>0</integer>
                        <key>generation</key>
                        <integer>1</integer>
                        <key>identifier</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3753</integer>
                        </dict>
                        <key>items</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3765</integer>
                        </dict>
                        <key>lightweightRequirement</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3764</integer>
                        </dict>
                        <key>modificationDate</key>
                        <real>744853746</real>
                        <key>name</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3761</integer>
                        </dict>
                        <key>programArguments</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3759</integer>
                        </dict>
                        <key>sha256</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3757</integer>
                        </dict>
                        <key>teamIdentifier</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3763</integer>
                        </dict>
                        <key>type</key>
                        <integer>8</integer>
                        <key>url</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3754</integer>
                        </dict>
                        <key>uuid</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>3752</integer>
                        </dict>
                </dict>
                <dict>
                        <key>$class</key>
                        <dict>
                                <key>CF$UID</key>
                                <integer>14</integer>
                        </dict>
                        <key>NS.uuidbytes</key>
                        <data>
                        zCtnIBzvRP2oy7/xv9P9Vw==
                        </data>
                </dict>
```

And if run `bgiparser --debug`, I get:

```
-------------------- Item: 172 --------------------
container: identifier "org.pqrs.Karabiner-Elements-Non-Privileged-Agents" and anchor apple generic and certificate 1[field.1.2.840.113635.100.6.2.6] /* exists */ and certificate leaf[field.1.2.840.113635.100.6.1.13] /* exists */ and certificate leaf[subject.OU] = G43BCU2T37
disposition: 11
url: {'NS.base': {'NS.base': '', 'NS.relative': 'file:///'}, 'NS.relative': 'Contents/Library/LaunchAgents/org.pqrs.service.agent.Karabiner-Menu.plist'}
items: []
developerName: Fumihiko Takayama
associatedBundleIdentifiers:
bundleIdentifier:
generation: 1
programArguments: ['/Library/Application Support/org.pqrs/Karabiner-Elements/Karabiner-Menu.app/Contents/MacOS/Karabiner-Menu']
uuid: 21B9CEFC-29BD-4F87-BE07-5A39DF5E6D9C
executablePath: /Library/Application Support/org.pqrs/Karabiner-Elements/Karabiner-Menu.app/Contents/MacOS/Karabiner-Menu
name: Karabiner-Menu
sha256: b'\xa9\x8bw\xe8\xcf\x86\xf7_:,\x01\x06\x11]<\xdd}\xce\xf5\xc3\xc2\xa30t\x81p\x86d\xdeg\xa9\xc5'
teamIdentifier: G43BCU2T37
type: 0x8
modificationDate: 744853746.0
flags: 0
identifier: org.pqrs.service.agent.Karabiner-Menu
executableModificationDate: 744853749.0
```

You still need me to email the entire file?

**Comment by mnrkbys** [comment]
Thanks. I have fixed (probably).
Please try [fix_sha256 branch](https://github.com/mnrkbys/bgiparser/tree/fix_sha256).

**Comment by huyz** [comment]
@mnrkbys Works, thanks!

**Comment by mnrkbys** [comment]
Merged #2
